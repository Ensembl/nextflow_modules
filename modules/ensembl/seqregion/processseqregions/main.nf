// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process SEQREGION_PROCESSSEQREGIONS {
    tag "${meta.species}"
    label 'process_low'
    
    conda "${moduleDir}/environment.yml"
    container "ensemblorg/ensembl-genomio:v1.6.0"

    input:
        tuple val (meta),
            path (genome_json),
            path (assembly_report),
            path (genomic_gbff)

    output:
        tuple val(meta), path("seq_region.json"), emit: seq_region
        path "versions.yml", emit: versions

    when:
        task.ext.when == null || task.ext.when

    script:
        output = "seq_region.json"
        """
        seq_region_prepare \
            --genome_file ${genome_json} \
            --report_file ${assembly_report} \
            --gbff_file ${genomic_gbff} \
            --dst_file ${output}
        
        schemas_json_validate --json_file ${output} --json_schema seq_region

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        seq_region_prepare --version >> versions.yml
        """

    stub:
        output = "seq_region.json"
        """
        seq_region_prepare \
            --genome_file ${genome_json} \
            --report_file ${assembly_report} \
            --gbff_file ${genomic_gbff} \
            --dst_file ${output} \
            --mock_run

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        seq_region_prepare --version >> versions.yml
        """
}
