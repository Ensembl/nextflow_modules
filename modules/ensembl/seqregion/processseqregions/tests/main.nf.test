// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test seqregion/processseqregions
nextflow_process {

    name "Test Process SEQREGION_PROCESSSEQREGIONS"
    script "../main.nf"
    process "SEQREGION_PROCESSSEQREGIONS"

    tag "seqregion"
    tag "seqregion/processseqregions"

    test("Process sequence regions [stub]") {

        when {

            options "-stub"

                process {
                    """
                    input[0] = [[ test_id:'dummy_core_db_core_110_1', species:'Genus species'],
                    file("${projectDir}/tests/modules/ensembl/seqregion/processseqregions/genome.json", checkIfExists: true),
                    file("${projectDir}/tests/modules/ensembl/seqregion/processseqregions/assembly_report.txt", checkIfExists: true),
                    file("${projectDir}/tests/modules/ensembl/seqregion/processseqregions/genome.gbff", checkIfExists: true)]
                    """
                }
            }

            then {
                assert snapshot(
                    process.out.seq_region.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("seq_region")

            assertAll(
                    { assert process.success },
                    { assert snapshot(process.out).match() },
                    { assert process.out.seq_region.size() == 1 },
                    { assert process.out.seq_region != null }
            )
        }
    }
}