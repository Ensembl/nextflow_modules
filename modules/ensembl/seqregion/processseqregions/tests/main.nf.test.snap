{
    "Process sequence regions [stub]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "test_id": "dummy_core_db_core_110_1",
                            "species": "Genus species"
                        },
                        "seq_region.json:md5,4fb69d15937c22072713730969dbbb48"
                    ]
                ],
                "1": [
                    "versions.yml:md5,86fd6f11a95c40520d76a6bb1d0362c6"
                ],
                "seq_region": [
                    [
                        {
                            "test_id": "dummy_core_db_core_110_1",
                            "species": "Genus species"
                        },
                        "seq_region.json:md5,4fb69d15937c22072713730969dbbb48"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,86fd6f11a95c40520d76a6bb1d0362c6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-19T16:09:51.012068117"
    },
    "seq_region": {
        "content": [
            [
                [
                    {
                        "test_id": "dummy_core_db_core_110_1",
                        "species": "Genus species"
                    },
                    "seq_region.json"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-19T11:58:02.018977677"
    }
}